src/util/psbt/map/global.rs

```diff
@@ -155,6 +250,55 @@ impl Decodable for Global {
                                 return Err(Error::InvalidKey(pair.key).into())
                             }
                         }
+                        PSBT_GLOBAL_KEY_XPUB => {
+                            if !pair.key.key.is_empty() {
+                                let xpub = ExtendedPubKey::decode(&pair.key.key)
+                                    .map_err(|_| {
+                                        encode::Error::ParseFailed("Can't deserialize ExtendedPublicKey from global XPUB key data")
+                                    })?;
+
+                                if pair.value.len() % 4 != 0 {
+                                    return Err(encode::Error::ParseFailed("Incorrect length of global xpub list"))
+                                }
+
+                                let keys_count = pair.value.len() / 4 - 1;
+                                let mut decoder = Cursor::new(pair.value);
+                                let mut fingerprint = [0u8; 4];
+                                decoder.read_exact(&mut fingerprint[..])?;
```

Comment:
nit: I think you can actually do
```rust
let mut fingerprint = Fingerprint::default();
decoder.read_exact(&mut fingerprint[..])?;
```
but I'm not 100% sure

Reply:
You can't: there is no `IndexMut` impl for `Fingerprint`